plugins/serializers/msgpack/msgpack.go

```diff
@@ -0,0 +1,51 @@
+package msgpack
+
+import (
+	"github.com/influxdata/telegraf"
+)
+
+// Serializer encodes metrics in MessagePack format
+type Serializer struct {
+}
+
+// NewSerializer creates a msgpack.Serializer
+func NewSerializer() *Serializer {
+	return &Serializer{}
+}
+
+func marshalMetric(buf []byte, metric telegraf.Metric) ([]byte, error) {
+	return (&Metric{
+		Name:   metric.Name(),
+		Time:   MessagePackTime{time: metric.Time()},
+		Tags:   metric.Tags(),
+		Fields: metric.Fields(),
+	}).MarshalMsg(buf)
+}
+
+// Serialize implements serializers.Serializer.Serialize
+// github.com/influxdata/telegraf/plugins/serializers/Serializer
+func (s *Serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	buf, err := marshalMetric(nil, metric)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// SerializeBatch implements serializers.Serializer.SerializeBatch
+// github.com/influxdata/telegraf/plugins/serializers/Serializer
+func (s *Serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+	buf := make([]byte, 0)
+	for _, m := range metrics {
+		var err error
+		buf, err = marshalMetric(buf, m)
+
+		if err != nil {
+			return nil, err
+		}
```

Comment:
In case you have multiple metrics you would just die and take all the metrics with you. Instead I propose to print an error message and continue. So we only use the one metric that cannot be serialized instead of losing all...

Reply:
Do you know which way is the protocol? I can not find the protocol from the interface code. 

https://github.com/influxdata/telegraf/blob/master/plugins/serializers/registry.go#L40-L43

Reply:
I think this is undefined. I created an issue regarding this. https://github.com/influxdata/telegraf/issues/8830

I'm fine with either way or we can add a config flag like `batch_serialize_ignore_errors` later. How does it sound to you?

Reply:
No please don't add a config flag. I stumbled upon this problem earlier and multiple times... The question is a bit, do you expect metrics to fail while others won't _or_ is it unlikely and in case of failure something is fundamentally broken. If the latter, please keep the code as is. Otherwise change to logging + `continue`.

Reply:
If we add the guard for the supported date range, it could fail partially. Otherwise, I don't have any example yet.

Reply:
Ok then leave it as you have it now because if the date really passes this limit more metrics will likely fail. In this case there is no point in doing the extra work.
Please mark as resolved.